environments/servlet/core/src/main/java/org/jboss/weld/environment/servlet/WeldServletLifecycle.java

```diff
@@ -278,10 +280,12 @@ protected CDI11Deployment createDeployment(ServletContext context, CDI11Bootstra
         final TypeDiscoveryConfiguration typeDiscoveryConfiguration = bootstrap.startExtensions(extensions);
         final EEModuleDescriptor eeModule = new EEModuleDescriptorImpl(context.getContextPath(), ModuleType.WEB);
 
-        final DiscoveryStrategy strategy = DiscoveryStrategyFactory.create(resourceLoader, bootstrap,
-                typeDiscoveryConfiguration.getKnownBeanDefiningAnnotations());
-        strategy.registerHandler(new ServletContextBeanArchiveHandler(context));
-
+        final DiscoveryStrategy strategy = DiscoveryStrategyFactory.create(resourceLoader, bootstrap, typeDiscoveryConfiguration.getKnownBeanDefiningAnnotations());
+        if (Reflections.isClassLoadable(resourceLoader, JANDEX_INDEX_CLASS_NAME)) {
+            strategy.registerHandler(new JandexServletContextBeanArchiveHandler(context));
```

Comment:
You shoul instantiate JandexServletContextBeanArchiveHandler using reflection...